### ISSUE TITLE
Lint checks cause third party lint checks to fail to load

### ISSUE THREAD

**Issue by mattmook** [uber/AutoDispose#396]
**Library version**: 1.4.0

**Repro steps**:

There's a bug in the current version of lint 26.5.1 (and earlier) when lint rules are defined using the service loader style. See https://issuetracker.google.com/issues/143455360

This buggy code can be avoided if instead the lint jar defines a manifest pointing to the IssueRegistry as follows in it's build.gradle file rather than using AutoService which could then be removed.

```
jar {
    manifest {
        attributes("Lint-Registry-v2": "com.uber.autodispose.lint.AutoDisposeIssueRegistry")
    }
}
```

**Comment by ZacSweers** [comment]
This isn't really an autodispose issue, nor do I think we should remove a feature just to work around another tool's bugs

**Comment by mattmook** [comment]
Arguably the library is implementing these lint rules in a non-standard, undocumented way - at least I can find no documentation any more that suggests adding lint rules through a service loader.

All the current documentation for custom lint rules use manifest entries, and this has been the case since at least 2015 (see http://tools.android.com/tips/lint-custom-rules, https://github.com/googlesamples/android-custom-lint-rules/blob/master/android-studio-3/checks/build.gradle,

**Comment by ZacSweers** [comment]
It is a standard way and was initially announced at KotlinConf 2017. I'm not sure why that repo doesn't use it, you should ask there :). Lint APIs in general are still technically in beta if that's what you mean, though by extension so is loading them via jar. Unfortunately lint documentation in general is fairly out dated, and much of it is more tribal knowledge or written in the lint-dev Google group.

Please pursue this in the bug you already filed above on AGP and accept our answer here. It does look like a legitimate issue. It also suggests that projects must be either all jar manifests or all service loaders, in which case removing the service loader implementation like you're asking would break users in the latter category.

**Comment by mattmook** [comment]
You are totally right, of course, I even went to that talk!

**Comment by ZacSweers** [comment]
Wow that was a quick turnaround from them! Will be fixed in 3.6
